File: Golang/Valid_Anagram.go
func isAnagram(s string, t string) bool {
	if len(s) != len(t) {
		return false
	}

	count := make([]int, 26)

	for _, char := range s {
		count[char-'a']++
	}

	for _, char := range t {
		count[char-'a']--
		if count[char-'a'] < 0 {
			return false
		}
	}
	return true
}

File: Kotlin/Valid_Anagram.kt
class Solution {
    fun isAnagram(s: String, t: String): Boolean {
        if (s.length != t.length) return false
        
        val count = IntArray(26)
        
        for (c in s) {
            count[c - 'a']++
        }
        
        for (c in t) {
            count[c - 'a']--
            if (count[c - 'a'] < 0) {
                return false
            }
        }
        return true
    }
}
